### DISCUSSION TITLE
Defining schedules with archetypal

### DISCUSSION THREAD

**Discussion by mormegil108** [samuelduchesne/archetypal#483] [Q&A] [answered]
Hello,

I am trying to create a UMI template using archetypal. Everything is fine except for the weekly schedules. I used two commands given on the documentatiton page (see https://archetypal.readthedocs.io/en/stable/creating_umi_template.html#defining-schedules). However, I keep getting the error: "TypeError: 'NoneType' object is not iterable" Can you help me with this one?  I upload the relevant notebook file.

[week_schedule.pdf](https://github.com/samuelduchesne/archetypal/files/12139924/week_schedule.pdf)

**Comment by samuelduchesne** [comment]
Hi! I'd love to help. Could you share the .ipynb notebook instead of the PDf? Thank you.

**Reply by mormegil108** [reply]
[week_schedule.zip](https://github.com/samuelduchesne/archetypal/files/12140100/week_schedule.zip)

**Reply by samuelduchesne** [reply, marked as answer]
Thank you!

Apologies, the documentation is a little bit misleading. It will be fixed soon.

The creation of schedules (and all other UMI objects) should not involve dictionaries; it is handled under the hood by `archetypal` when a UMI Template Library is serialized to a JSON file.

I corrected the instructions here; the notebook should run as is: 
[scratch_9.ipynb.zip](https://github.com/samuelduchesne/archetypal/files/12140116/scratch_9.ipynb.zip)

**Comment by mormegil108** [comment]
Hey Samuel,

Thank you for your help. I appreciate it.

Best,
Said
